.github/workflows/sda-pipeline.yml

```diff
@@ -0,0 +1,119 @@
+      run: bash ./dev_tools/scripts/deploy-oidc.sh
+    - name: Deploy minio
+      if: matrix.test == 's3-inbox'
+      run: bash ./dev_tools/scripts/deploy-minio.sh
+    - name: Wait for minio to become ready
+      if: matrix.test == 's3-inbox'
+      run: bash ./dev_tools/scripts/wait-for-pods.sh minio
+    - name: debug minio
+      if: failure() && matrix.test == 's3-inbox'
+      run: kubectl describe pod -l app=minio
+    - name: log minio
+      if: failure() && matrix.test == 's3-inbox'
+      run: kubectl logs $(kubectl get pods | grep minio | awk '{print $1}')
+    - name: Start CEGA services
+      run: bash ./dev_tools/scripts/deploy-cega.sh
+    - name: Wait for CEGA to become ready
+      run: bash ./dev_tools/scripts/wait-for-pods.sh cega-mq
+    - name: debug cega
+      if: failure()
+      run: |
+        for c in cega-users cega-mq
+        do kubectl describe pod -l app=$c
+        done
+    - name: log cega
+      if: failure()
+      run: |
+        for c in cega-users cega-mq
+        do kubectl logs $(kubectl get pods | grep $c | awk '{print $1}')
+        done
+    - name: Deploy SDA message broker
+      run: bash ./dev_tools/scripts/deploy-mq.sh
+    - name: Wait for broker to become ready
+      run: bash ./dev_tools/scripts/wait-for-pods.sh broker
+    - name: debug MQ
+      if: failure()
+      run: kubectl describe pod -l role=broker
+    - name: log MQ
+      if: failure()
+      run: kubectl logs broker-sda-mq-0
+    - name: Deploy the SDA stack for posix
+      if: matrix.test == 'sftp-inbox'
+      run: |
+        kubectl apply -f .github/ci_tests/posix-volumes.yaml;
+        bash ./dev_tools/scripts/deploy-sda-posix.sh;
+    - name: Wait for posix SDA stack to become ready
+      if: matrix.test == 'sftp-inbox'
+      run: bash ./dev_tools/scripts/wait-for-pods.sh "${{ env.posix_svc_list }}"
+    - name: debug posix sda
+      if: failure() && matrix.test == 'sftp-inbox'
+      run: |
+        for s in ${{ env.posix_svc_list }}
+        do kubectl describe pod -l role=$s
+        done
+    - name: log posix sda
+      if: failure() && matrix.test == 'sftp-inbox'
+      run: |
+        for s in ${{ env.posix_svc_list }}
+        do kubectl logs $(kubectl get pods | grep $s | awk '{print $1}')
+        done
+    - name: Deploy the SDA stack for s3
+      if: matrix.test == 's3-inbox'
+      run: bash ./dev_tools/scripts/deploy-sda-s3.sh
+    - name: Wait for s3 SDA stack to become ready
+      if: matrix.test == 's3-inbox'
+      run: bash ./dev_tools/scripts/wait-for-pods.sh "${{ env.s3_svc_list }}"
+    - name: debug s3 sda
+      if: failure() && matrix.test == 's3-inbox'
+      run: |
+        for s in ${{ env.s3_svc_list }}
+        do kubectl describe pod -l role=$s
+        done
+    - name: log s3 sda
+      if: failure() && matrix.test == 's3-inbox'
+      run: |
+        for s in ${{ env.s3_svc_list }}
+        do kubectl logs $(kubectl get pods | grep $s | awk '{print $1}')
+        done
```

Comment:
This should not display logs from all services but rather from the one that was failing.
It's really annoying to have to scroll past 500+ lines of logs to get to the one that has the information that is relevant for the issue.